src/main/scala/util/PlusArg.scala

```diff
@@ -20,6 +22,42 @@ object PlusArg
   // Add a docstring to document the arg, which can be dumped in an elaboration
   // pass.
 
-  def apply(name: String, default: Int = 0, docstring: String = ""): UInt =
+  def apply(name: String, default: Int = 0, docstring: String = ""): UInt = {
+    PlusArgArtefacts.append(name, default, docstring)
     Module(new plusarg_reader(name + "=%d", default, docstring)).io.out
+  }
+}
+
+object PlusArgArtefacts {
+  private var artefacts: Map[String, PlusArgInfo] = Map.empty
+
+  /* Add a new PlusArg */
+  def append(name: String, default: Int, docstring: String): Unit =
+    artefacts = artefacts ++ Map(name -> PlusArgInfo(default, docstring))
+
+  /* From plus args, generate help text */
+  private def serializeHelp(tab: String = ""): String = artefacts
+    .map{ case(arg, PlusArgInfo(default, docstring)) =>
+      s"""|$tab+$arg=INT\\n\\
+          |$tab${" "*20}$docstring\\n\\
+          |$tab${" "*22}(default=$default)""".stripMargin }.toSeq
+    .mkString("\\n\\\n") ++ "\""
+
+  /* From plus args, generate a char array of their names */
+  private def serializeArray(tab: String = ""): String = {
```

Comment:
can you name this function to indicate that it's generating C code? `serializeArrayC` and `serializeC` for the one below? (or whatever you think is appropriate naming)